W-22497227 Agent Fabric Overview (#431)
mulesoft/docs-general: modules/ROOT/pages/agent-fabric-overview.adoc

* Role in Agent Fabric: MCP Bridge converts existing APIs automatically through configuration. MCP Connector is for teams that need to build custom MCP server logic, compose multiple systems, apply transformation, orchestrate interactions with external MCP services, or expose capabilities that don’t map directly to a single existing API.

* What you can do:
** Build Mule applications that expose tools and resources to MCP clients
** Connect Mule applications to external MCP servers
** Compose multiple backend systems into a single MCP server
** Define custom logic for how agent requests are handled
** Orchestrate workflows that incorporate AI agents and external MCP services
* When you need it: Use MCP Connector when you need to expose capabilities to agents that require custom logic, multi-system composition, orchestration with external MCP services, or transformation that configuration-based tools like MCP Bridge don’t cover.

> * Learn more: xref:connectors::mcp::index.adoc[MCP Connector]

Reviewer: ```suggestion
* Learn more: xref:mcp-connector::index.adoc[MCP Connector]
```

Reviewer: I don’t think these links will resolve correctly, so please update all connector links globally to use this format across MCP Connector, A2A Connector, and Other AI Connectors sections

MuleSoft Agent Fabric helps you turn agent sprawl into a governed, coordinated, intelligent network. Regardless of the platform where your AI agents were built or are running, Agent Fabric provides the infrastructure to discover, orchestrate, govern, and observe agents across your enterprise. Agent Fabric enables you to build intelligent agent networks where multiple agents work together, access enterprise systems securely, and operate under consistent governance and observability.

[[how-agent-fabric-works]]
== How MuleSoft Agent Fabric Works

Every organization has systems, agents, LLMs, and data sources scattered across platforms and teams. Left unmanaged, this creates agent sprawl: siloed agents that can't work together, inconsistent experiences, security gaps, and costs that are difficult to control. Agent Fabric solves this by providing a unified control plane for the agentic era.

Agent Fabric starts with enterprise actionability, bridging the gap between your existing enterprise stack and the agentic world. It makes every system, agent, LLM, and data source directly usable in agentic workflows without rebuilding them, giving the entire fabric a foundation to build on.

> MuleSoft supports two complementary products that can be used together for agentic enterprises. Each serves a distinct role in building and managing your organization's AI agent strategy:

Reviewer 2: I thought agent fabric wasn't a product?

* Agent Fabric
+
The agent control plane for managing and governing agents across any platform, including Salesforce Agentforce, Amazon Bedrock, Google Vertex AI, Microsoft Copilot Studio, and others. Use Agent Fabric when you need to discover, catalog, and govern agents built anywhere, orchestrate multi-vendor workflows where agents from different platforms must work together, or apply consistent security, compliance, and cost controls across your entire agent ecosystem.

Agentforce and Agent Fabric are core components of Salesforce's Agentic Enterprise Architecture, a four-layer platform model designed for the era of AI agents and headless operations. This architecture separates the data foundation, business logic, orchestration, and experience layers. 

Agent Fabric operates in the orchestration layer, providing the control plane for multi-vendor agent coordination.

image::agent-fabric-pillars.png[Agent Fabric pillars, discover, govern, orchestrate, and observe]

As the agent control plane for your enterprise, Agent Fabric delivers four core capabilities that work together, whether you're managing Agentforce agents, third-party agents, or both:

> * <<discover,Discover>>

Reviewer 2: should these be Discovery, Governance, Orchestration, and Observability?

Author: I aligned those to match the graphic.

+
A central registry, supported with automated discovery, where every agent and MCP server — regardless of where it's built — can be cataloged, discovered, and reused. This eliminates duplication and makes it easy to compose solutions from existing assets.
* <<govern,Govern>>
+
Enterprise-grade guardrails ensuring every agent, MCP server, and LLM interaction is secure, compliant, and cost-controlled, so teams can innovate with confidence.
* <<orchestrate,Orchestrate>>
+

> The ability to coordinate agents and tools across ecosystems with guided determinism, so multi-step processes run smoothly and every agent works together to deliver reliable business outcomes.

Reviewer 2: ```suggestion
The coordination of agents and tools across ecosystems with guided determinism, so multi-step processes run smoothly and every agent works together to deliver reliable business outcomes.
```

Reviewer 2: I think this makes it more parallel with the way the other definitions are written

* What you can do:
** Connect LLMs and vector databases to agentic workflows without custom integration
** Unlock existing AI infrastructure for use inside agent networks
** Enrich agent reasoning with organizational knowledge stores

* When you need it: Use AI Connectors when your agents need to reason with LLMs or retrieve context from vector databases your organization already operates.

* Learn more:
** xref:agentforce-connector::index.adoc[]
** xref:amazon-bedrock-connector::index.adoc[]
** xref:einstein-ai-connector::index.adoc[]
** xref:mulesoft-ai-chain-connector::index.adoc[]
** xref:mulesoft-inference-connector::index.adoc[]
** xref:mulesoft-vectors-connector::index.adoc[]

=== B2B (Partner Manager)

* What it is: Anypoint Partner Manager enables bidirectional Business-to-Business message exchanges with external trading partner ecosystems through configuration-driven integration. Partner Manager handles technical partner onboarding, message flow orchestration, and transaction monitoring across multiple transport protocols (AS2, SFTP, HTTP/S, FTP) and message formats (X12, EDIFACT, CSV, JSON, XML).

> * Role in Agent Fabric: Partner Manager makes B2B trading partner networks directly actionable for agents without requiring them to understand complex EDI standards or protocol-specific integration patterns. Agents can initiate B2B transactions, query partner message flows, monitor transmission status, and respond to partner interactions as part of multi-step workflows, turning manual B2B operations into automated, agent-driven processes.

Reviewer 2: ```suggestion
* Role in Agent Fabric: Partner Manager makes B2B trading partner networks directly actionable for agents without requiring them to understand complex EDI standards or protocol-specific integration patterns. Agents can initiate B2B transactions, query partner message flows, monitor transmission status, and respond to partner interactions as part of multi-step workflows, turning manual B2B operations into automated agent-driven processes.
```

* Role in Agent Fabric: Agent Visualizer makes the agent network legible, both during development, where you need to verify that routing logic and agent connections are configured correctly, and in production, where you need to identify bottlenecks, trace failures, and optimize performance. It turns multi-agent interactions into clear, interactive maps so you can trace decisions, monitor real-time health, and detect issues before they impact outcomes. Without this visibility, multi-agent systems are opaque and difficult to operate with confidence.
* Key capabilities:
** Visual network topology showing agents, brokers, MCP servers, and their connections
** Performance metrics including latency, throughput, and error rates
** Historical analysis for identifying patterns over time

* When you need it: Use Agent Visualizer during development to verify your network is wired correctly, at deployment to confirm traffic is routing as expected, and in production to troubleshoot.

* Learn more: xref:agent-visualizer::index.adoc[]

=== Monitor

> * What it is: In the enhanced MuleSoft experience, monitoring is available at both the service level and the organization level. At the service level, each agent, API, MCP server, LLM proxy, and gateway has a Monitoring tab showing runtime health signals and detailed time-series breakdowns, depending on your gateway, runtime path, and platform connections. At the organization level, Observability aggregates dashboards, reports, and alerts when your administrator has connected the observability backend for your business group.

Reviewer 2: should observability be capitalized?

** Compare signals across instances or environments to attribute changes to deployments or policy updates, allowing you to baseline behavior after changes.
** Configure alert notifications to route API and runtime alerts to Slack, email, or other channels your team monitors
** Access organization-wide dashboards and reports through Observability, using filters to narrow your investigation down to a specific service, environment, or route.

* When you need it: Use monitoring during development for performance validation, and continuously in production to ensure reliability, meet SLAs, and catch issues before they affect users.

* Learn more: xref:exp-services-monitoring.adoc[]

[[supporting-tools]]
== Supporting Tools

=== Anypoint Code Builder

* What it is: Anypoint Code Builder is the development environment for building agent networks, configuring agent interactions, and developing custom MCP servers.
* Role in Agent Fabric: Anypoint Code Builder is where agent networks come to life. It provides the visual editor canvas and tooling to define an agent network YAML and develop custom MCP servers, all before deploying to production.
* Key capabilities:

> ** Design and configure agent network structures with visual editors

Reviewer 2: the visual editor it read-only so this may be a bit misleading. Same with the statement above--"It provides the visual editor canvas..."

** Use MuleSoft Vibes to generate and validate network configurations from natural language
* Learn more: xref:anypoint-code-builder::index.adoc[]

=== MuleSoft Vibes

* What it is: MuleSoft Vibes is the AI assistant built into Anypoint Code Builder, trained on industry best practices for graph-based agent architecture.
* Role in Agent Fabric: Vibes accelerates the process of building agent networks by letting you describe what you want in natural language and generating robust broker and network configurations from those descriptions. Vibes validates that the resulting architecture is sound and aligned with best practices before you deploy.
* What you can do:
** Describe your desired agent network in natural language and generate YAML configurations
** Validate network structure and routing logic against architecture best practices
** Accelerate development from initial design through testing and refinement
* When you need it: Use MuleSoft Vibes throughout the development lifecycle, from initial network design through iterating on routing logic and agent configurations.
* Learn more: xref:anypoint-code-builder::vibes-get-started.adoc[]

== What's Next

> * Get started with the xref:learning-map-agent-fabric.adoc[] for a guided path through discovering assets, building your first agent network, and deploying it with governance and monitoring.

Reviewer 2: ```suggestion
* For a guided path through discovering assets, building your first agent network, and deploying it with governance and monitoring, get started with the xref:learning-map-agent-fabric.adoc[].
```

> * Explore the https://www.mulesoft.com/lp/ebook/mulesoft-agent-fabric-technical-overview[MuleSoft Agent Fabric Technical Overview] for a comprehensive guide to the technical foundations, capability pillars, and architectural best practices that power enterprise agentic transformation.

Reviewer 2: ```suggestion
* For a comprehensive guide to the technical foundations, capability pillars, and architectural best practices that power enterprise agentic transformation, explore the https://www.mulesoft.com/lp/ebook/mulesoft-agent-fabric-technical-overview[MuleSoft Agent Fabric Technical Overview].
```
